src/logcollector/read_multiline.c

```diff
@@ -93,7 +93,13 @@ void *read_multiline(logreader *lf, int *rc, int drop_it) {
             buffer_size++;
         }
 
-        strncpy(buffer + buffer_size, str, OS_MAXSTR - buffer_size - 2);
+        strncat(buffer, str, sizeof(buffer) - 1 - buffer_size);
```

Comment:
We could use snprintf here.

Reply:
We are substituting `strncpy` for `snprintf`. In this case, the `strncat` focus is different from `snprintf`.

Reply:
I meant we could use `snprintf` and pass `buffer + buffer_size` as the target buffer, the same way the code was doing it with `strncpy`. That way we avoid `strncat` having to go through the whole buffer to find the null terminator on each iteration of the for loop.